core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -148,6 +244,16 @@
      */
     public abstract long lastModified() throws IOException;
 
+    /**
+     * Gets the file’s Unix mode, if meaningful.
```

Comment:
If the file is Symlink, which mode should be returned by the implementation? Symlink or the linked file? I suspect the later one taking the abstraction, but makes sense to confirm

Reply:
`IOUtils.mode` currently calls `Files.getPosixFilePermissions` _without_ `LinkOption#NOFOLLOW_LINKS`, so the latter.  Prior to #3135, it used `stat`, which says

> If the named file is a symbolic link, the stat() function shall continue pathname resolution using the contents of the symbolic link, and shall return information pertaining to the resulting file if the file exists.

So I suppose yes, and I just need to document this.

Reply:
yes, just a minor doc update